### PR TITLE
Random.int added

### LINKED ISSUE #759: [Feature] Implement random.int function
**Is your feature request related to a problem? Please describe.**
Eldritch currently can't create a random integer.

**Describe the solution you'd like**

Implement the `random.int` function that will return a random (`i32`) integer. This will theoretically be very similar to the [`random.bool`](https://github.com/spellshift/realm/blob/main/implants/lib/eldritch/src/random/bool_impl.rs#L5) function. 

To see an example of what a PR that implements an eldritch function looks like, check out the PR for `sys.get_user()` [here](https://github.com/spellshift/realm/pull/257).

### PR DESCRIPTION
**Pull request by chase1k**
#### What type of PR is this?

/kind documentation
/kind feature
/kind eldritch-function

#### What this PR does / why we need it:

Adds random.int function to generate a secure random int

#### Which issue(s) this PR fixes:

Fixes #759

### CODE REVIEW

**Review by hulto — COMMENTED** [review]
Test

File: implants/lib/eldritch/src/random/int_impl.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    #[test]
+    fn test_random_int_uniform() -> anyhow::Result<()> {
+        let mut counts = vec![0; MAX_VALUE as usize];
+        for _ in 0..NUM_ITERATION {
+            let random_number = int(MIN_VALUE, MAX_VALUE)?;
+            counts[random_number as usize] += 1;
+        }
+
+        let lower_bound = 0.90 * NUM_ITERATION as f64 / MAX_VALUE as f64;
```
Comment (hulto) [inline implants/lib/eldritch/src/random/int_impl.rs:36]: I don't know math but seems okay.

**Review by hulto — DISMISSED** [review]
LGTM once tests pass

File: docs/_docs/user-guide/eldritch.md [hunk 2]
```diff
@@ -692,6 +692,12 @@ The random library is designed to enable generation of cryptogrphically secure r
 
 The <b>random.bool</b> method returns a randomly sourced boolean value.
 
+### radnom.int
```
Comment (Cictrone) [inline docs/_docs/user-guide/eldritch.md:695]: nit: spelling

**Review by Cictrone — CHANGES_REQUESTED** [review]
lgtm besides typo!

File: docs/_docs/user-guide/eldritch.md (hunk 2 above)
Comment (chase1k) [reply docs/_docs/user-guide/eldritch.md]: Fixed

**Review by jabbate19 — DISMISSED** [review]

**Review by hulto — DISMISSED** [review]
LGTM

**Review by Cictrone — APPROVED** [review]

**Pull request merged**